server/src/main/java/com/paypal/selion/grid/MobileProcessLauncher.java

```diff
@@ -0,0 +1,121 @@
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+import static com.paypal.selion.pojos.SeLionGridConstants.*;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.annotations.Beta;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.paypal.selion.logging.SeLionGridLogger;
+
+/**
+ * A {@link RunnableLauncher} for mobile WebDrivers (appium, selendroid, and ios-driver). Adds the ability to process
+ * default arguments for the mobile WebDrivers (appium, selendroid, and ios-driver) from the SeLion Grid JSON config
+ * file.
+ */
+@Beta
+class MobileProcessLauncher extends AbstractBaseProcessLauncher {
+    private static final SeLionGridLogger LOGGER = SeLionGridLogger.getLogger(MobileProcessLauncher.class);
+    private static final String SEPARATOR = "\n----------------------------------\n";
+    JsonObject defaultArgs;
+
+    public MobileProcessLauncher(String[] args) {
+        init(args);
+    }
+
+    public MobileProcessLauncher(String[] args, ProcessLauncherOptions options) {
+        init(args, options);
+    }
+    
+    /**
+     * Get program arguments to pass
+     *
+     * @return the program arguments to pass represented as an array of {@link String}
+     * @throws IOException
+     */
+    @Override
+    String[] getProgramArguments() throws IOException {
+        LOGGER.entering();
+        List<String> args = new LinkedList<String>();
+        args.addAll(Arrays.asList(super.getProgramArguments()));
+
+        // add the defaults which we don't already have a value for
+        Iterator<Entry<String, JsonElement>> iterator = defaultArgs.entrySet().iterator();
+        while (iterator.hasNext()) {
+            String key = iterator.next().getKey();
+            if (!args.contains(key)) {
+                args.add(key);
+                String val = defaultArgs.get(key).getAsString();
+                if (StringUtils.isNotBlank(val)) {
+                    args.add(val);
+                }
+            }
+        }
+
+        // filter out SeLion Grid specific args which do not apply
+        List<String> filteredArgs = new LinkedList<String>();
+        filteredArgs.add(SELION_CONFIG_ARG);
+        filteredArgs.add(SELION_NOCONTINUOS_ARG);
+
+        for (String filter : filteredArgs) {
+            if (!filter.equals(SELION_NOCONTINUOS_ARG)) {
+                args.remove(args.indexOf(filter) + 1);
+            }
+            args.remove(filter);
+        }
+
+        LOGGER.exiting(args.toString());
+        return args.toArray(new String[args.size()]);
+    }
+
+    void printUsageInfo() {
+        StringBuffer usage = new StringBuffer();
```

Comment:
@mach6 
I saw the same StringBuffer being used in a couple of other places as well. So can we just take care of all of that in one shot ? I forgot to call that out before.